runtime/bin/directory_linux.cc

```diff
@@ -117,20 +117,18 @@ ListType DirectoryListingEntry::Next(DirectoryListing* listing) {
 
   // Iterate the directory and post the directories and files to the
   // ports.
-  int status = 0;
-  dirent entry;
   dirent* result;
-  status = NO_RETRY_EXPECTED(readdir_r(
-      reinterpret_cast<DIR*>(lister_), &entry, &result));
-  if ((status == 0) && (result != NULL)) {
-    if (!listing->path_buffer().Add(entry.d_name)) {
+  errno = 0;
+  result = readdir(reinterpret_cast<DIR*>(lister_));
+  if ((errno == 0) && (result != NULL)) {
```

Comment:
`result != NULL` implies `errno == 0` according to the man page.

Only if `result == NULL` one needs to distinguish between error (in which case `errno != 0`) and end of stream (in which case `errno == 0`).